federation/cmd/federation-apiserver/app/autoscaling.go

```diff
@@ -21,24 +21,35 @@ import (
 	"k8s.io/apiserver/pkg/registry/generic"
 	"k8s.io/apiserver/pkg/registry/rest"
 	genericapiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/apiserver/pkg/server/storage"
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/apis/autoscaling"
 	_ "k8s.io/kubernetes/pkg/apis/autoscaling/install"
+	autoscalingv1 "k8s.io/kubernetes/pkg/apis/autoscaling/v1"
 	hpastorage "k8s.io/kubernetes/pkg/registry/autoscaling/horizontalpodautoscaler/storage"
 )
 
-func installAutoscalingAPIs(g *genericapiserver.GenericAPIServer, optsGetter generic.RESTOptionsGetter) {
-	hpaStorage, hpaStatusStorage := hpastorage.NewREST(optsGetter)
-
-	autoscalingResources := map[string]rest.Storage{
-		"horizontalpodautoscalers":        hpaStorage,
-		"horizontalpodautoscalers/status": hpaStatusStorage,
+func installAutoscalingAPIs(g *genericapiserver.GenericAPIServer, optsGetter generic.RESTOptionsGetter, apiResourceConfigSource storage.APIResourceConfigSource) {
+	groupName := autoscaling.GroupName
+	hpaStorageFn := func() map[string]rest.Storage {
```

Comment:
Yes it does need to be a func, but I wrote it that way because I had some future refactoring in mind where I want this to be shared between kubernetes and federation. A single function that returns the storage for a resource (for ex storages for hpa and its subresources in this case).
I didnt do that refactoring in this PR but I do have plans for a follow up PR.

So if you think this is bad, then I can change this to inline for now and will change them to funcs later.
But if its fine then I will like to keep them as is. Let me know.